packages/eslint-plugin/src/rules/no-duplicate-type-union-intersection-members.ts

```diff
@@ -0,0 +1,122 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import * as util from '../util';
+
+export type Options = [
+  {
+    checkIntersections?: boolean;
+    checkUnions?: boolean;
+  },
+];
+export type MessageIds = 'duplicate' | 'suggestFix';
+
+export default util.createRule<Options, MessageIds>({
+  name: 'no-duplicate-type-union-intersection-members',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow duplicate union/intersection type members',
+      recommended: false,
+    },
+    fixable: 'code',
+    hasSuggestions: true,
+    messages: {
+      duplicate: '{{type}} type member {{name}} is duplicated.',
+      suggestFix: 'Delete duplicated members of type (removes all comments).',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          checkIntersections: {
+            type: 'boolean',
+          },
+          checkUnions: {
+            type: 'boolean',
+          },
+        },
+      },
+    ],
+  },
+  defaultOptions: [
+    {
+      checkIntersections: true,
+      checkUnions: true,
```

Comment:
IME it's confusing for folks to have the default value for something be truthy. How about switching these to `ignoreIntersections` & `ignoreUnions`?

Reply:
Exactly! 
This rule check both intersection and union duplication by default, so I think `ignore` option would be clearer!